evm_interpreter/src/instructions/host.rs

```diff
@@ -236,7 +236,9 @@ impl<'ee, S: EthereumLikeTypes> Interpreter<'ee, S> {
         self.resize_heap(mem_offset, len)?;
         let data = &self.heap[mem_offset..mem_offset + len];
 
-        tracer.on_event(THIS_EE_TYPE, &self.address, &topics, data);
+        if tracer.is_on_event_enabled() {
```

Comment:
I don't think this is something that needs to be toggled on and off during tracer execution, so consider using `TracerType::IS_ON_EVENT_ENABLED` instead.

Reply:
Personally, I don't find the readability advantage of `impl` notation to be a reason not to use explicit type params.

Technically, you could avoid type params by adding optional methods to the trait that perform the if.

Reply:
Hmm, sounds interesting. Ok, I'll try to change it to generics after all

Reply:
Did you try and see that the separate `is_enabled` is necessary? Explicit generics or impl trait should make no difference for inlining.

Reply:
Seems like it works as expected now!